scheme/process-context.scm

```diff
@@ -0,0 +1,34 @@
+(define-library (scheme process-context)
+
+   (import
+      (owl defmac)
+      (only (owl primop) halt)
+      (only (owl syscall) error)
+      (only (owl sys) getenv)
+      (scheme base))
+
+   (export
+      command-line
+      emergency-exit
+      exit
+      get-environment-variable
+      get-environment-variables)
+
+   (begin
+      (define (command-line)
```

Comment:
These can be implemented by having having a thread to store the varying content. For example, one could have a ff in a scheme state thread started at repl startup along with other base threads, command line arguments could be sent there and (command-line) would turn into something like (interact 'scheme-state (cons 'get 'command-line-args)).